<a href="https://colab.research.google.com/github/PedroHudson/Codigos_Python_aleatorios/blob/main/C%C3%B3pia_de_PYTHON_Lista_de_Exerc%C3%ADcios_do_Curso_em_V%C3%ADdeos.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Exercício Python 1: Escreva a frase 'Olá, mundo'.

In [ ]:
print('Olá, mundo')


In [ ]:
msg = 'Olá, Mundo'
pergunta = 'Como vai você?'
print(msg + '. ' + pergunta)


Exercício Python 2: Faça um programa que leia o nome de uma pessoa e mostre uma mensagem de boas-vindas.

In [ ]:
nome = input('Qual é o seu nome? ')

print('É um prazer te conhecer, ' + nome)


In [ ]:
nome = input('Qual é o seu nome? ')
print('É um prazer te conhecer,{}!'.format(nome))


Exercício Python 3: Crie um programa que leia dois números e mostre a soma entre eles.

In [ ]:
valor1 = int(input('Digite um valor:'))
valor2 = int(input('Digite outro valor:'))
soma = valor1 + valor2
print('A some desses valores é igual a {}!'.format(soma))

In [ ]:
valor1 = int(input('Digite um número: '))
valor2 = int(input('Digite um número, mais uma vez: '))
soma = valor1+valor2
print('A some de {} e de {} é igual a {}.'.format(valor1, valor2, soma))


Exercício Python 4: Faça um programa que leia algo pelo teclado e mostre na tela o seu tipo primitivo e todas as informações possíveis sobre ele.

In [ ]:
frase = input('Digite algo: ')
print('O tipo primitivo dessa frase é', type(frase))
print('É somente espaço? ', frase.isspace())
print('É número? ', frase.isnumeric())
print('É alfanúmerico?', frase.isalpha())
print('Está em maiúsculas? ', frase.isupper())
print('Está em minúsculas? ', frase.islower())
print('Está capitalizada? ', frase.istitle())


ordem de procesdência das operações aritiméticas

1 ( )

2 **

3 * / // %

4 + -

Exercício Python 5: Faça um programa que leia um número Inteiro e mostre na tela o seu sucessor e seu antecessor.

In [ ]:
numero = int(input('Digite um número: '))
antecessor = numero - 1
sucessor = numero + 1
print('Você digitou o {}, o antecessor de {} é {} e o sucessor é {}!'.format(numero, numero, antecessor, sucessor))



In [ ]:
numero = int(input('Digite um número: '))
print('Você digitou o número {}, o antecessor é {} e o sucessor é o {}!'.format(numero, (numero-1), (numero+1)))


Exercício Python 006: Crie um algoritmo que leia um número e mostre o seu dobro, triplo e raiz quadrada.


In [ ]:
numero = int(input('Digite um número: '))
print('Você digitou o número', numero)
print('O dobro de {} é {}'.format(numero, (numero*2)))
print('O triplo de {} é {}'.format(numero, (numero*3)))
print('A raiz quadrada de {} é {}'.format(numero, (numero**2)))


Exercício Python 7: Desenvolva um programa que leia as duas notas de um aluno, calcule e mostre a sua média.

In [ ]:
nota1 = float(input('Qual foi sua nota? '))
nota2 = float(input('Qual foi sua outra nota? '))
print('Sua média é {}'.format((nota1+nota2)/2))


Exercício Python 8: Escreva um programa que leia um valor em metros e o exiba convertido em centímetros e milímetros.


In [ ]:
valor_em_metros = float(input('Digite o valor em metros: '))
print('{}m é igual a {:.0f}cm e {:.0f}mm.'.format(valor_em_metros, (valor_em_metros*100), (valor_em_metros*1000)))


Exercício Python 9: Faça um programa que leia um número Inteiro qualquer e mostre na tela a sua tabuada.


In [ ]:
num = int(input('Digite um número: '))
print(f'{num} x 1 = {num*1}\n')
print(f'{num} x 2 = {num*2}\n')
print(f'{num} x 3 = {num*3}\n')
print(f'{num} x 4 = {num*4}\n')
print(f'{num} x 5 = {num*5}\n')
print(f'{num} x 6 = {num*6}\n')
print(f'{num} x 7 = {num*7}\n')
print(f'{num} x 8 = {num*8}\n')
print(f'{num} x 9 = {num*9}\n')
print(f'{num} x 10 = {num*10}\n')

Digite um número: 4
4 x 1 = 4

4 x 2 = 8

4 x 3 = 12

4 x 4 = 16

4 x 5 = 20

4 x 6 = 24

4 x 7 = 28

4 x 8 = 32

4 x 9 = 36

4 x 10 = 40



In [ ]:
num = int(input('Digite um número: '))
c = 0
for c in range(1,11):
  print(f'{num}x{c} = {num*c}')
print('Fim.')

Digite um número: 5
5x1 = 5
5x2 = 10
5x3 = 15
5x4 = 20
5x5 = 25
5x6 = 30
5x7 = 35
5x8 = 40
5x9 = 45
5x10 = 50
Fim.


Exercício Python 10: Crie um programa que leia quanto dinheiro uma pessoa tem na carteira e mostre quantos dólares ela pode comprar.

In [ ]:
dinheiro = float(input('Quanto dinheiro você tem? '))
cotacao_dolar = 4.77

print(f'Você consegue comprar U${dinheiro/cotacao_dolar:.2f}')

Quanto dinheiro você tem? 70003
Você consegue comprar U$14675.68


Exercício Python 11: Faça um programa que leia a largura e a altura de uma parede em metros, calcule a sua área e a quantidade de tinta necessária para pintá-la, sabendo que cada litro de tinta pinta uma área de 2 metros quadrados.


In [ ]:
largura = int(input('Digite a largura da parede: '))
altura = int(input('Digite a altura da parede: '))
area = largura*altura
litro = 2

print(f'A área da parede é {area} metros quadrados. Você precisa de {area/litro} litros para pintar a parede')


Digite a largura da parede: 20
Digite a altura da parede: 12
A área da parede é 240 metros quadrados. Você precisa de 120.0 litros para pintar a parede


Exercício Python 12: Faça um algoritmo que leia o preço de um produto e mostre seu novo preço, com 5% de desconto.


In [ ]:
preco = float(input('Digite o preço: '))
desconto = preco-(preco*0.05)

print(f'Era R$ {preco}. Agora custa apenas R${desconto}')


Digite o preço: 100
Era R$ 100.0. Agora custa apenas R$95.0


Exercício Python 13: Faça um algoritmo que leia o salário de um funcionário e mostre seu novo salário, com 15% de aumento.


In [ ]:
salario = float(input('Digite o valor do seu salário: '))
aumento = salario * 0.15
novo_salario = salario + aumento
print(f'Seu novo salário é de R$ {novo_salario}')


Digite o valor do seu salário: 5000
Seu novo salário é de R$ 5750.0


Exercício Python 14: Escreva um programa que converta uma temperatura
 digitando em graus Celsius e converta para graus Fahrenheit.

In [ ]:
c = float(input('Temperature in Celcius: '))
f = ((9/5) * c) + 32
print(f)

Temperature in Celcius: 38
100.4


In [ ]:
print('*~'*30)
print('TEMPERATURE CONVERSOR'.center(60))
print('*~'*30)

conversor = int(input('Select: \n[1] CELCIUS TO FARENHEIT\n[2] FARENHEIT TO CELCIUS\n\n'))

if conversor == 1:
  c = float(input('Temperature in Celcius: '))
  f = ((9/5) * c) + 32
  print(f'{c} celcius = {f:.2f} farenheit.')

elif conversor == 2:
  f = float(input('Temperature in farenheit: '))
  c = (5/9) * (f - 32)
  print(f'{f} farenheit = {c:.2f} celcius.')

else:
  print('Você digitou a opção errada.')





*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~
                   TEMPERATURE CONVERSOR                    
*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~
Select: 
[1] CELCIUS TO FARENHEIT
[2] FARENHEIT TO CELCIUS

2
Temperature in farenheit: 100
100.0 farenheit = 37.78 celcius.


In [ ]:
from time import sleep
print('*~'*30)
print('TEMPERATURE CONVERSOR'.center(60))
print('*~'*30)
choice = ' '

while True:
  conversor = int(input('Select: \n[1] CELCIUS TO FARENHEIT\n[2] FARENHEIT TO CELCIUS\n\n'))
  if conversor == 1:
    c = float(input('Temperature in Celcius: '))
    f = ((9/5) * c) + 32
    print(f'{c} celcius = {f:.1f} farenheit.')

  elif conversor == 2:
    f = float(input('Temperature in farenheit: '))
    c = (5/9) * (f - 32)
    print(f'{f} farenheit = {c:.1f} celcius.')

  else:
    print('This option is invalid.')
  choice  = input('Continue?\nSelect: \n[Y] YES \n[N] NO\n\n').strip().upper()
  if choice == 'N':
     break
  elif choice != 'Y':
    print('This option is invalid.')

sleep(2)
print('END')

*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~
                   TEMPERATURE CONVERSOR                    
*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~*~
Select: 
[1] CELCIUS TO FARENHEIT
[2] FARENHEIT TO CELCIUS

1
Temperature in Celcius: 100
100.0 celcius = 212.0 farenheit.
Continue?
Select: 
[Y] YES 
[N] NO

y
This option is invalid.
Select: 
[1] CELCIUS TO FARENHEIT
[2] FARENHEIT TO CELCIUS

2
Temperature in farenheit: 100
100.0 farenheit = 37.8 celcius.
Continue?
Select: 
[Y] YES 
[N] NO

n
END


**Exercício** Python 15: Escreva um programa que pergunte a quantidade de Km percorridos por um carro alugado e a quantidade de dias pelos quais ele foi alugado. Calcule o preço a pagar, sabendo que o carro custa R$ 60  por dia e R$ 0,15 por Km rodado.

Exercício Python 16: Crie um programa que leia um número Real qualquer pelo teclado e mostre na tela a sua porção Inteira.

In [ ]:
from math import trunc
numero_real = float(input('Digite um número real: '))
print('Você digitou o número {} e o número inteiro é {}! '.format(numero_real, trunc(numero_real)))


Digite um número real: 5
Você digitou o número 5.0 e o número inteiro é 5! 


In [ ]:
numero_real = float(input('Digite um número real: '))
print('Você digitou {} e o número real é {}'.format(numero_real, int(numero_real)))


Exercício Python 17: Faça um programa que leia o comprimento do cateto oposto e do cateto adjacente de um triângulo retângulo. Calcule e mostre o comprimento da hipotenusa.

*A soma do quadrado dos catetos é igual a raiz quadrada da hipotenusa. \*

In [ ]:
cateto = float(input('Digite um número: '))
cateto_adjacente = float(input('Digite outro número: '))
print('A hipotenusa é {:.2f}.'.format((cateto **2 + cateto_adjacente **2) ** (1/2)))



In [ ]:
from math import hypot
cateto = float(input('Digite um número: '))
cateto_adjacente = float(input('Digite outro número: '))
print('A hipotenusa é {:.2f}!'.format(hypot(cateto, cateto_adjacente)))

Exercício Python 18: Faça um programa que leia um ângulo qualquer e mostre na tela o valor do seno, cosseno e tangente desse ângulo.

In [ ]:
import math

# Solicitar um ângulo em graus
angulo_graus = float(input('Digite um ângulo em graus: '))

# Converter o ângulo para radianos
angulo_radianos = math.radians(angulo_graus)

# Calcular seno, cosseno e tangente
seno = math.sin(angulo_radianos)
cosseno = math.cos(angulo_radianos)
tangente = math.tan(angulo_radianos)

# Exibir os resultados
print(f'Seno: {seno:.2f}')
print(f'Cosseno: {cosseno:.2f}')
print(f'Tangente: {tangente:.2f}')


Digite um ângulo em graus: 45
Seno: 0.71
Cosseno: 0.71
Tangente: 1.00


Exercício Python 19: Um professor quer sortear um dos seus quatro alunos para apagar o quadro. Faça um programa que ajude ele, lendo o nome dos alunos e escrevendo na tela o nome do escolhido.

In [ ]:
from random import choice
nome1 = str(input('Digite o primeiro nome: '))
nome2 = str(input('Digite o segundo nome: '))
nome3 = str(input('Digite o tereiro nome: '))
nome4 = str(input('Digite o quarto nome: '))
nome_dos_alunos = [nome1, nome2, nome3, nome4]
print('O aluno escolhido foi: {}'.format(choice(nome_dos_alunos)))

Exercício Python 20: O mesmo professor do desafio 19 quer sortear a ordem de apresentação de trabalhos dos alunos. Faça um programa que leia o nome dos quatro alunos e mostre a ordem sorteada.

In [ ]:
from random import shuffle
nome1 = input('Digite o primeiro nome: ')
nome2 = input('Digite o segundo nome: ')
nome3 = input('Digite o terceiro nome: ')
nome4 = input('Digite o quarto nome: ')
lista_dos_alunos = [nome1, nome2, nome3, nome4]
print('A ordem de aprentação será: {}'.format(shuffle(lista_dos_alunos)))

In [ ]:
from random import shuffle
nome1 = input('Digite o primeiro nome: ')
nome2 = input('Digite o segundo nome: ')
nome3 = input('Digite o terceiro nome: ')
nome4 = input('Digite o quarto nome: ')
lista_dos_alunos = [nome1, nome2, nome3, nome4]
shuffle(lista_dos_alunos)
print('A ordem de aprentação será: {}'.format(lista_dos_alunos))


Exercício Python 21: Faça um programa em Python que abra e reproduza o áudio de um arquivo MP3.

In [ ]:
import pygame
pygame.init()
pygame.mixer.music.load('/content/wonderwomanogg.ogg')
pygame.mixer.music.play()
input(pygame.event.wait())



Exercício Python 22: Crie um programa que leia o nome completo de uma pessoa e mostre:

– O nome com todas as letras maiúsculas e minúsculas.

– Quantas letras ao todo (sem considerar espaços).

In [ ]:
nome = str(input('Digite seu nome completo: '))
print(format(nome.upper()))
print(format(nome.lower()))
print(format(len(nome)))

Digite seu nome completo: Clara
CLARA
clara
5


Exercício Python 23: Faça um programa que leia um número de 0 a 9999 e mostre na tela cada um dos dígitos separados.

In [ ]:
# Solicita ao usuário que insira um número de 0 a 9999
numero = int(input('Digite um número entre 0 e 9999: '))

# Obtém cada dígito separadamente
unidade = numero % 10
dezena = (numero // 10) % 10
centena = (numero // 100) % 10
milhar = (numero // 1000) % 10

# Exibe os dígitos na tela
print(f'Milhar: {milhar}')
print(f'Centena: {centena}')
print(f'Dezena: {dezena}')
print(f'Unidade: {unidade}')

Digite um número entre 0 e 9999: 523
Milhar: 0
Centena: 5
Dezena: 2
Unidade: 3


Exercício Python 24: Crie um programa que leia o nome de uma cidade diga se ela começa ou não com o nome “SANTO”.

In [ ]:
cidade = input('Digite o nome da cidade:').strip()
print((cidade[:5]).upper() == 'SANTO')

Digite o nome da cidade:Santo Antonio
True


Exercício Python 25: Crie um programa que leia o nome de uma pessoa e diga se ela tem “SILVA” no nome.

In [ ]:
sobrenome = input('Digite o seu nome completo:').strip().upper()
print(f'Tem Silva no nome? {"SILVA" in sobrenome}')

Digite o seu nome completo:clara silva
Tem Silva no nome? True


*Exercício* Python 26: Faça um programa que leia uma frase pelo teclado e mostre quantas vezes aparece a letra “A”, em que posição ela aparece a primeira vez e em que posição ela aparece a última vez.

Exercício Python 27: Faça um programa que leia o nome completo de uma pessoa, mostrando em seguida o primeiro e o último nome separadamente.

In [ ]:
# Solicita ao usuário que insira uma frase
frase = input('Digite uma frase: ')

# Converte a frase para maiúsculas para tornar a contagem insensível a maiúsculas/minúsculas
frase_maiuscula = frase.upper()

# Conta quantas vezes a letra 'A' aparece na frase
quantidade_a = frase_maiuscula.count('A')

# Encontra a posição da primeira ocorrência de 'A'
primeira_ocorrencia = frase_maiuscula.find('A') + 1  # Adiciona 1 para ajustar para a indexação começando em 1

# Encontra a posição da última ocorrência de 'A'
ultima_ocorrencia = frase_maiuscula.rfind('A') + 1  # Adiciona 1 para ajustar para a indexação começando em 1

# Exibe os resultados
print(f'A letra "A" aparece {quantidade_a} vezes na frase.')
print(f'A primeira ocorrência de "A" está na posição {primeira_ocorrencia}.')
print(f'A última ocorrência de "A" está na posição {ultima_ocorrencia}.')


Exercício Python 28: Escreva um programa que faça o computador “pensar” em um número inteiro entre 0 e 5 e peça para o usuário tentar descobrir qual foi o número escolhido pelo computador. O programa deverá escrever na tela se o usuário venceu ou perdeu.

In [ ]:
##não funciona com o choice
from random import choice
escolha = int(input('Escolha um número de 0 a 5: '))
numeros = [0, 1, 2, 3, 4, 5]
print('O número sorteado foi {}'.format(choice(numeros)))
if escolha == (choice(numeros)):
  print('PARABÉNS, VOCÊ VENCEU!')
else:
  print('VOCÊ PERDEU. TENTE OUTRA VEZ')


Escolha um número de 0 a 5: 4
O número sorteado foi 5
VOCÊ PERDEU. TENTE OUTRA VEZ


In [ ]:
from random import randint
computador = randint(0,5) ## o computador vai escolher um número de 0 a 5.
aposta = int(input('Escolha um número de 0 a 5.' ))
print('O número sorteado foi {}!'.format(computador))
if aposta == computador:
  print('Você ganhou 10 mil.')
print('Você perdeu. Tente novamente.')


Escolha um número de 0 a 5.4
O número sorteado foi 2!
Você perdeu. Tente novamente.


Exercício Python 29: Escreva um programa que leia a velocidade de um carro. Se ele ultrapassar 80Km/h, mostre uma mensagem dizendo que ele foi multado. A multa vai custar R$7,00 por cada Km acima do limite.

In [ ]:
velocidade = int(input('Velocidade atual: '))
radar = 80
multa = 7
if velocidade < radar:
  print('Velocidade atual: {}km/h, boa viagem!'.format(velocidade))
else:
  print('ATENÇÃO! Velocidade atual: {}km/h, você foi MULTADO em R${},00.'.format(velocidade, (velocidade-80)*7))

Velocidade atual: 82
ATENÇÃO! Velocidade atual: 82km/h, você foi MULTADO em R$14,00.


Exercício Python 30: Crie um programa que leia um número inteiro e mostre na tela se ele é PAR ou ÍMPAR.

In [ ]:
numero = int(input('Digite um número: '))
resto = numero % 2
if resto == 0:
  print('Esse número é par')
else:
  print('Esse número é impar')


Digite um número: 3
Esse número é impar


Exercício Python 31: Desenvolva um programa que pergunte a distância de uma viagem em Km. Calcule o preço da passagem, cobrando R$0,50 por Km para viagens de até 200Km e R$0,45 parta viagens mais longas.



In [ ]:
distancia = int(input('Qual será a distância percorrida em Km? '))
if distancia <= 200:
  print('A distância estimada é de {} Km, você pagará R${} pela viagem.'.format(distancia, (distancia * (50/100))))
else:
  print('A distância estimada é de {} Km, você pagará R${} pela viagem.'.format(distancia, (distancia * (45/100))))

Qual será a distância percorrida em Km? 210
A distância estimada é de 210 Km, você pagará R$94.5 pela viagem.



Exercício Python 32: Faça um programa que leia um ano qualquer e mostre se ele é bissexto.

Exercício Python 33: Faça um programa que leia três números e mostre qual é o maior e qual é o menor.

Exercício Python 34: Escreva um programa que pergunte o salário de um funcionário e calcule o valor do seu aumento. Para salários superiores a R$1250,00, calcule um aumento de 10%. Para os inferiores ou iguais, o aumento é de 15%.

Exercício Python 35: Desenvolva um programa que leia o comprimento de três retas e diga ao usuário se elas podem ou não formar um triângulo.

Exercício Python 36: Escreva um programa para aprovar o empréstimo bancário para a compra de uma casa. Pergunte o valor da casa, o salário do comprador e em quantos anos ele vai pagar. A prestação mensal não pode exceder 30% do salário ou então o empréstimo será negado.

In [ ]:
casa = float(input('VALOR DO IMÓVEL: '))
salario = float(input('SALÁRIO LÍQUIDO: '))
anos = int(input('PERÍODO DO PAGAMENTO (ANOS)'))
periodo = (anos)*12
prestacao = casa/periodo
percentual = salario * 0.30

print(f'LIMTE: {percentual}')

if prestacao > percentual:
  print(f'EMPRESTIMO NEGADO. PRESTAÇÕES: {prestacao:.2f} ACIMA DO LIMITE.')

else:
  print(f'EMPRESTIMO APROVADO. PRESTAÇÕES: {prestacao:.2f}.')

VALOR DO IMÓVEL: 100000
SALÁRIO LÍQUIDO: 5000
PERÍODO DO PAGAMENTO (ANOS)15
LIMTE: 1500.0
EMPRESTIMO APROVADO. PRESTAÇÕES: 555.56.


Exercício Python 37: Escreva um programa em Python que leia um número inteiro qualquer e peça para o usuário escolher qual será a base de conversão: 1 para binário, 2 para octal e 3 para hexadecimal.


Exercício Python 038: Escreva um programa que leia dois números inteiros e compare-os. mostrando na tela uma mensagem:

– O primeiro valor é maior

– O segundo valor é maior

– Não existe valor maior, os dois são iguais


In [ ]:
a = int(input('Digite um número inteiro: '))
b = int(input('Digite outro número inteiro: '))

if a > b:
  print(f'O número {a} é maior que {b}.')
elif b > a:
  print(f'O número {b} é maior que {a}.')
else:
  print('Não há diferença entre os números.')

Digite um número inteiro: 5
Digite outro número inteiro: 2
O número 5 é maior que 2.


Exercício Python 39: Faça um programa que leia o ano de nascimento de um jovem e informe, de acordo com a sua idade, se ele ainda vai se alistar ao serviço militar, se é a hora exata de se alistar ou se já passou do tempo do alistamento. Seu programa também deverá mostrar o tempo que falta ou que passou do prazo.

Exercício Python 040: Crie um programa que leia duas notas de um aluno e calcule sua média, mostrando uma mensagem no final, de acordo com a média atingida:

– Média abaixo de 5.0: REPROVADO

– Média entre 5.0 e 6.9: RECUPERAÇÃO

– Média 7.0 ou superior: APROVADO

In [ ]:
media1 = float(input('Qual a sua primeira nota? ' ))
media2 = float(input('Qual a sua segunda nota? '))
media_final = (media1 + media2) / 2
if media_final <= 5:
  print('Sua média final é {}, você foi REPROVADO.'.format(media_final))
elif media_final >= 7:
  print('Sua média final é {}, você foi APROVADO.'.format(media_final))
else:
  print('Sua média final é {}, você ficou em RECUPERAÇÃO.'.format(media_final))

Qual a sua primeira nota? 5
Qual a sua segunda nota? 4
Sua média final é 4.5, você foi REPROVADO.


Exercício Python 041: A Confederação Nacional de Natação precisa de um programa que leia o ano de nascimento de um atleta e mostre sua categoria, de acordo com a idade:

– Até 9 anos: MIRIM

– Até 14 anos: INFANTIL

– Até 19 anos: JÚNIOR

– Até 25 anos: SÊNIOR

– Acima de 25 anos: MASTER


Exercício Python 42: Refaça o DESAFIO 35 dos triângulos, acrescentando o recurso de mostrar que tipo de triângulo será formado:

– EQUILÁTERO: todos os lados iguais

– ISÓSCELES: dois lados iguais, um diferente

– ESCALENO: todos os lados diferentes

Exercício Python 43: Desenvolva uma lógica que leia o peso e a altura de uma pessoa, calcule seu Índice de Massa Corporal (IMC) e mostre seu status, de acordo com a tabela abaixo:

– IMC abaixo de 18,5: Abaixo do Peso

– Entre 18,5 e 25: Peso Ideal

– 25 até 30: Sobrepeso

– 30 até 40: Obesidade

– Acima de 40: Obesidade Mórbida

In [ ]:
altura = float(input('Qual a sua altura em Cm? '))
peso = float(input('Qual o seu peso em Kg? '))
imc = peso / altura ** 2
print('Seu IMC é {:.2f}.'.format(imc))
if imc <= 18.5:
  print('Você está abaixo do peso!')
elif imc > 18.5 and imc <= 25:
  print('Você está no peso ideal.')
elif imc > 25 and imc <= 30:
  print('Você está com sobrepeso.')
elif imc > 30 and imc <= 40:
  print('Você está com obsidade.')
elif imc > 40:
  print('Você está com obesidade mórbida.')

Qual a sua altura em Cm? 1.70
Qual o seu peso em Kg? 120
Seu IMC é 41.52.
Você está com obesidade mórbida.


Exercício Python 44: Elabore um programa que calcule o valor a ser pago por um produto, considerando o seu preço normal e condição de pagamento:

– à vista dinheiro/cheque: 10% de desconto

– à vista no cartão: 5% de desconto

– em até 2x no cartão: preço formal

– 3x ou mais no cartão: 20% de juros

Exercício Python 45: Crie um programa que faça o computador jogar Jokenpô com você.

In [ ]:
from random import randint
from time import sleep

opcoes = ['PEDRA', 'PAPEL', 'TESOURA']
escolha_computador = randint(0,2)

print ('<<<<JO KEN PO>>>>\n')

print ('Escolha sua opção: \n [0] = PEDRA \n [1] = PAPEL \n [2] = TESOURA ')

escolha_usuario = int(input('Qual a sua opção: '))

print ('=o'*11)

print (f'Você escolheu: {opcoes[escolha_usuario]}\n')

print ('JO  ✊')
sleep(1)
print ('KEN  ✊')
sleep(1)
print ('PO  ✊')
sleep(1)
print (f'\nO computador escolheu: {opcoes[escolha_computador]}')

print ('=o'*11)

##simbolos ✊✋✌
if escolha_computador == 0: ##✊
  if escolha_usuario ==  1:
    print('✊ < ✋ \nVocê VENCEU! ✴✧✦')
  elif escolha_usuario == 2:
    print('✊ > ✌ \nVocê PERDEU! ✴✧✦')
  elif escolha_usuario == 0:
    print('✊ > ✊ \nEMPATOU! ✴✧✦')
  else:
    print ('ESCOLHA INVÁLIDA')
elif escolha_computador == 1: ## ✋
  if escolha_usuario == 0:
    print ('✋ < ✊ \nVocê PERDEU! ✴✧✦')
  elif escolha_usuario == 2:
    print ('✋ < ✌ \nVocê VENCEU! ✴✧✦')
  elif escolha_usuario == 1:
    print('✋ > ✋ \nEMPATOU! ✴✧✦')
  else:
    print ('ESCOLHA INVÁLIDA')

elif escolha_computador == 2: ##✌
  if escolha_usuario == 0:
    print ('✌ < ✊ \nVocê VENCEU! ✴✧✦')
  elif escolha_usuario == 1:
    print('✌ < ✋ \nVocê PERDEU! ✴✧✦')
  elif escolha_usuario == 2:
    print('✌ > ✌ \nEMPATOU! ✴✧✦')
  else:
    print ('ESCOLHA INVÁLIDA')


<<<<JO KEN PO>>>>

Escolha sua opção: 
 [0] = PEDRA 
 [1] = PAPEL 
 [2] = TESOURA 
Qual a sua opção: 1
=o=o=o=o=o=o=o=o=o=o=o
Você escolheu: PAPEL

JO  ✊
KEN  ✊
PO  ✊

O computador escolheu: PAPEL
=o=o=o=o=o=o=o=o=o=o=o
✋ > ✋ 
EMPATOU! ✴✧✦


In [ ]:
import random

def jogar_jokenpo(escolha_usuario):
    opcoes = ['PEDRA', 'PAPEL', 'TESOURA']
    escolha_computador = random.choice(opcoes)

    print(f'Você escolheu: {escolha_usuario}')
    print(f'O computador escolheu: {escolha_computador}')

    if escolha_usuario == escolha_computador:
        resultado = 'Empate!'
    elif (escolha_usuario == 'PEDRA' and escolha_computador == 'TESOURA') or \
         (escolha_usuario == 'PAPEL' and escolha_computador == 'PEDRA') or \
         (escolha_usuario == 'TESOURA' and escolha_computador == 'PAPEL'):
        resultado = 'Você ganhou!'
    else:
        resultado = 'Você perdeu!'

    print(resultado)

# Solicita ao usuário que faça uma escolha
escolha_usuario = input('Escolha entre PEDRA, PAPEL ou TESOURA: ').upper()

# Verifica se a escolha é válida
if escolha_usuario in ['PEDRA', 'PAPEL', 'TESOURA']:
    jogar_jokenpo(escolha_usuario)
else:
    print('Escolha inválida. Por favor, escolha entre PEDRA, PAPEL ou TESOURA.')


Exercício Python 46: Faça um programa que mostre na tela uma contagem regressiva para o estouro de fogos de artifício, indo de 10 até 0, com uma pausa de 1 segundo entre eles.

In [ ]:
from time import sleep
for c in range(10,-1, -1):
  print(c)
  sleep(1)
print('BOOM!')

10
9
8
7
6
5
4
3
2
1
0
BOOM!


Exercício Python 47: Crie um programa que mostre na tela todos os números pares que estão no intervalo entre 1 e 50.



In [ ]:
for par in range(1,51):
  if par % 2 == 0:
    print (par)

2
4
6
8
10
12
14
16
18
20
22
24
26
28
30
32
34
36
38
40
42
44
46
48
50


In [ ]:
for par in range(2,51,2):
  print(par)

2
4
6
8
10
12
14
16
18
20
22
24
26
28
30
32
34
36
38
40
42
44
46
48
50


Exercício Python 48: Faça um programa que calcule a soma entre todos os números que são múltiplos de três e que se encontram no intervalo de 1 até 500.

In [ ]:
soma = 0

for num in range(1, 501):
    if num % 3 == 0:
        soma += num

print("A soma de todos os números múltiplos de três no intervalo de 1 a 500 é:", soma)


A soma de todos os números múltiplos de três no intervalo de 1 a 500 é: 41583


Exercício Python 49: Refaça o DESAFIO 9, mostrando a tabuada de um número que o usuário escolher, só que agora utilizando um laço for.



In [ ]:
numero = int(input('Digite um número:'))
for c in range(1,11):
  print(f'{numero}x{c} = {numero*c}')

Digite um número:3
3x1 = 3
3x2 = 6
3x3 = 9
3x4 = 12
3x5 = 15
3x6 = 18
3x7 = 21
3x8 = 24
3x9 = 27
3x10 = 30


Exercício Python 50: Desenvolva um programa que leia seis números inteiros e mostre a soma apenas daqueles que forem pares. Se o valor digitado for ímpar, desconsidere-o.

In [ ]:
soma = 0
c = 0
for n in range(1,7):
  num = int(input('Digite um número:'))
  if num % 2 == 0:
    soma += num
    c += 1
print(f'Foram o total de {c} numeros pares e a soma é igual a {soma}.')



Digite um número:1
Digite um número:2
Digite um número:3
Digite um número:4
Digite um número:5
Digite um número:6
Foram o total de 3 numeros pares e a soma é igual a 12.


Exercício Python 51: Desenvolva um programa que leia o primeiro termo e a razão de uma PA. No final, mostre os 10 primeiros termos dessa progressão.

In [ ]:
print ('PROGRESSÃO ARITIMÉTICA')

termo = int(input('Primeiro Termo:'))
razao = int(input('Razão:'))
pa = 0
for n in range(0,10):
  pa = (termo+n)*razao
  print(pa)

print('Acabou!')

PROGRESSÃO ARITIMÉTICA
Primeiro Termo:0
Razão:2
0
2
4
6
8
10
12
14
16
18
Acabou!


Exercício Python 52: Faça um programa que leia um número inteiro e diga se ele é ou não um número primo.**negrito**

In [ ]:
num = int(input("Digite um número inteiro: "))

# Um número primo é maior que 1 e divisível apenas por 1 e por ele mesmo
if num > 1:
    # Verifica se o número é divisível por qualquer número entre 2 e ele mesmo - 1
    for i in range(2, num):
        if num % i == 0:
            print(num, "não é um número primo.")
            break
    else:
        print(num, "é um número primo.")
else:
    print(num, "não é um número primo.")


Digite um número inteiro: 7
7 é um número primo.


Exercício Python 53: Crie um programa que leia uma frase qualquer e diga se ela é um palíndromo, desconsiderando os espaços. Exemplos de palíndromos:


In [ ]:
frase = input('Escreva uma frase: ').replace(' ','').upper()
palindromo = frase[::-1]
if frase == palindromo:
  print('Isso é um palindromo')
else:
  print('Frase normal.')

Escreva uma frase: o lobo ama o bolo
Isso é um palindromo


Exercício Python 53: Crie um programa que leia uma frase qualquer e diga se ela é um palíndromo, desconsiderando os espaços. Exemplos de palíndromos:

In [ ]:
frase = str(input('Escreva uma frase: ')).replace(' ','').upper()
for f in range(len(frase)-1, -1, -1):
  p = frase[::-1]
if frase == p:
  print(f'A frase {frase} é um palíndromo.')
else:
  print('A frase é normal.')

Escreva uma frase: Hannah
A frase HANNAH é um palíndromo.


In [ ]:
def eh_palindromo(frase):
    # Remover espaços da frase e converter para minúsculas
    frase = frase.replace(" ", "").lower()

    # Verificar se a frase é igual à sua inversa
    if frase == frase[::-1]:
        return True
    else:
        return False

frase = input("Digite uma frase: ")

if eh_palindromo(frase):
    print("A frase é um palíndromo!")
else:
    print("A frase não é um palíndromo.")


Exercício Python 54: Crie um programa que leia o ano de nascimento de sete pessoas. No final, mostre quantas pessoas ainda não atingiram a maioridade e quantas já são maiores.



```
# Isto está formatado como código
```



In [ ]:
ano = 2024
maior = 0
menor = 0
invalida = 0
for n in range(1,8):
  data = int(input(f'Digite a {n}ª data de nascimento : '))
  id = ano - data
  if id > 18:
    maior += 1
  elif id < 18:
    menor += 1
  else:
    invalida +=1

print(f'Foram {menor} menor(es), {maior} maior(es) e {invalida} inválida(s)')


Digite a 1ª data de nascimento : 2018
Digite a 2ª data de nascimento : 2002
Digite a 3ª data de nascimento : 2005
Digite a 4ª data de nascimento : 2019
Digite a 5ª data de nascimento : 2000
Digite a 6ª data de nascimento : 1985
Digite a 7ª data de nascimento : 1982
Foram 2 menor(es), 5 maior(es) e 0 inválida(s)


Exercício Python 55: Faça um programa que leia o peso de cinco pessoas. No final, mostre qual foi o maior e o menor peso lidos.

*texto em itálico*

In [ ]:
maior_peso = menor_peso = 0
for n in range(1,6):
  peso = float(input(f'Digite o peso da {n}ª pessoa: '))
  if n == 1:
     maior_peso = menor_peso = peso
  else:

    if peso > maior_peso:
       maior_peso = peso
    elif peso < menor_peso:
      menor_peso = peso

print(f"O maior peso lido fon {maior_peso} kg.")
print(f"O menor peso lido foi {menor_peso} kg.")



Digite o peso da 1ª pessoa: 55
Digite o peso da 2ª pessoa: 57
Digite o peso da 3ª pessoa: 85
Digite o peso da 4ª pessoa: 82
Digite o peso da 5ª pessoa: 88
O maior peso lido fon 88.0 kg.
O menor peso lido foi 55.0 kg.


Exercício Python 56: Desenvolva um programa que leia o nome, idade e sexo de 4 pessoas. No final do programa, mostre: a média de idade do grupo, qual é o nome do homem mais velho e quantas mulheres têm menos de 20 anos.



In [ ]:
soma_idade = 0
maior_idade_homem = 0
nome_homem_mais_velho = ''
contagem_mulheres_menos_20 = 0

for pessoa in range(1, 5):
    print(f'--- {pessoa}ª pessoa ---')
    nome = input('Nome: ')
    idade = int(input('Idade: '))
    sexo = input('Sexo [M/F]: ').strip().upper()

    soma_idade += idade

    if sexo == 'M' and idade > maior_idade_homem:
        maior_idade_homem = idade
        nome_homem_mais_velho = nome

    if sexo == 'F' and idade < 20:
        contagem_mulheres_menos_20 += 1

media_idade = soma_idade / 4

print(f'\nA média de idade do grupo é de {media_idade:.1f} anos.')
print(f'O homem mais velho se chama {nome_homem_mais_velho} e tem {maior_idade_homem} anos.')
print(f'{contagem_mulheres_menos_20} mulheres têm menos de 20 anos.')


Exercício Python 57: Faça um programa que leia o sexo de uma pessoa, mas só aceite os valores ‘M’ ou ‘F’. Caso esteja errado, peça a digitação novamente até ter um valor correto.

In [ ]:
print('Sexo:\n M = MASCULINO\n F = FEMININO\n O = OUTRO')
sexo = input('Informe seu sexo? ').strip().upper()
while sexo not in ['F', 'M', 'O']:
  sexo = input('Dados inválidos. Informe seu sexo novamente? ').strip().upper()
if sexo == 'M':
  sexo = 'Masculino'
elif sexo == 'F':
  sexo = 'Feminino'
elif sexo == 'O':
  sexo = 'Outro'

print(f'Dado registrado com sucesso. Usuário do sexo {sexo}.')

Sexo:
 M = MASCULINO
 F = FEMININO
 O = OUTRO
Informe seu sexo? t
Dados inválidos. Informe seu sexo novamente? g
Dados inválidos. Informe seu sexo novamente? f
Dado registrado com sucesso. Usuário do sexo Feminino.


Exercício Python 58: Melhore o jogo do DESAFIO 28 onde o computador vai “pensar” em um número entre 0 e 10. Só que agora o jogador vai tentar adivinhar até acertar, mostrando no final quantos palpites foram necessários para vencer.



```
# Isto está formatado como código
```



In [ ]:
from random import choice
palpite = int(input('Escolha um número de 0 a 10: '))
numeros = range(0,11)
numero_sorteado = (choice(numeros))
palpites = 1
while palpite not in [numero_sorteado]:
  if palpite < numero_sorteado:
    print ('O número é maior.')
  else:
    print('O número é menor.')
  palpite = int(input('Tente novamente: '))
  palpites += 1

print (f'O número sorteado foi {numero_sorteado}. Você acertou após {palpites} jogada(s)!')


Escolha um número de 0 a 10: 3
O número é maior.
Tente novamente: 4
O número é maior.
Tente novamente: 6
O número é maior.
Tente novamente: 8
O número é maior.
Tente novamente: 9
O número sorteado foi 9. Você acertou após 5 jogada(s)!


In [ ]:
from random import randint
computador = randint(0,10)
print('Sou eu computador. Acabei de pensar em um número.')
print('Será que você consegue adivinhar qual foi?')
acertou = False
palpites = 0

while not acertou:
  jogador = int(input('Qual o seu palpite? '))
  palpites += 1
  if jogador == computador:
    acertou = True
  else:
    if jogador < computador:
      print('Mais...Tente outra vez.')
    elif jogador > computador:
      print('Menos... Tente outra vez.')
print (f'Acertou com {palpites} tentativas.')



Sou eu computador. Acabei de pensar em um número.
Será que você consegue adivinhar qual foi?
Qual o seu palpite?10
Menos... Tente outra vez.
Qual o seu palpite?5
Mais...Tente outra vez.
Qual o seu palpite?6
Mais...Tente outra vez.
Qual o seu palpite?8
Mais...Tente outra vez.
Qual o seu palpite?9
Acertou com 5 tentativas.


Exercício Python 059: Crie um programa que leia dois valores e mostre um menu na tela:

[ 1 ] somar

[ 2 ] multiplicar

[ 3 ] maior

[ 4 ] novos números

[ 5 ] sair do programa

Seu programa deverá realizar a operação solicitada em cada caso.

In [ ]:
valor_1 = float(input('Digite um valor: '))
valor_2 = float(input('Digite um valor: '))
print('Escolha qual operação você quer realizar.')
operacao = int(input('[ 1 ] somar\n[ 2 ] multiplicar\n[ 3 ] maior\n[ 4 ] novos números\n[ 5 ] sair do programa  '))
while operacao not in [5]:
  if operacao == 1:
    print(f'{valor_1} + {valor_2} = {valor_1+valor_2}')
  elif operacao == 2:
    print(f'{valor_1} x {valor_2} = {valor_1*valor_2}')
  elif operacao == 3:
    if valor_1 > valor_2:
      print(f"{valor_1} é maior que {valor_2}.")
    elif valor_2 > valor_1:
      print(f"{valor_2} é maior que {valor_1}.")
    else:
      print(f"{valor_2} é igual a {valor_1}.")
  elif operacao == 4:
    print('Digite novos valores.')
    valor_1 = float(input('Digite um valor: '))
    valor_2 = float(input('Digite um valor: '))
    operacao = int(input('[ 1 ] somar\n[ 2 ] multiplicar\n[ 3 ] maior\n[ 4 ] novos números\n[ 5 ] sair do programa  '))
  elif operacao == 5:
    print('Você saiu do programa.')
  else:
    print('Operação inválida.')
  print('Escolha qual operação você quer realizar.')
  operacao = int(input('[ 1 ] somar\n[ 2 ] multiplicar\n[ 3 ] maior\n[ 4 ] novos números\n[ 5 ] sair do programa  '))
print('Fim do programa')




Digite um valor: 2
Digite um valor: 3
Escolha qual operação você quer realizar.
[ 1 ] somar
[ 2 ] multiplicar
[ 3 ] maior
[ 4 ] novos números
[ 5 ] sair do programa  1
2.0 + 3.0 = 5.0
Escolha qual operação você quer realizar.
[ 1 ] somar
[ 2 ] multiplicar
[ 3 ] maior
[ 4 ] novos números
[ 5 ] sair do programa  2
2.0 x 3.0 = 6.0
Escolha qual operação você quer realizar.
[ 1 ] somar
[ 2 ] multiplicar
[ 3 ] maior
[ 4 ] novos números
[ 5 ] sair do programa  3
3.0 é maior que 2.0.
Escolha qual operação você quer realizar.
[ 1 ] somar
[ 2 ] multiplicar
[ 3 ] maior
[ 4 ] novos números
[ 5 ] sair do programa  4
Digite novos valores.
Digite um valor: 1
Digite um valor: 2
[ 1 ] somar
[ 2 ] multiplicar
[ 3 ] maior
[ 4 ] novos números
[ 5 ] sair do programa  3
Escolha qual operação você quer realizar.
[ 1 ] somar
[ 2 ] multiplicar
[ 3 ] maior
[ 4 ] novos números
[ 5 ] sair do programa  5
Fim do programa


Exercício Python 060: Faça um programa que leia um número qualquer e mostre o seu fatorial. Exemplo:

5! = 5 x 4 x 3 x 2 x 1 = 120

In [ ]:
from math import factorial
valor = int(input('Digite um número inteiro: '))
c = valor
f = factorial(valor)
while c > 0:
  print(f'{c}', end = '')
  print(' x ' if c > 1 else ' = ', end = '')
  c -= 1
print(f)

Digite um número inteiro: 5
5 x 4 x 3 x 2 x 1 = 120


Exercício Python 61: Refaça o DESAFIO 51, lendo o primeiro termo e a razão de uma PA, mostrando os 10 primeiros termos da progressão usando a estrutura while.


-
*Exercício Python 51: Desenvolva um programa que leia o primeiro termo e a razão de uma PA. No final, mostre os 10 primeiros termos dessa progressão. *

In [ ]:
print ('PROGRESSÃO ARITIMÉTICA')

termo = int(input('Primeiro Termo:'))
razao = int(input('Razão:'))
t = termo
c = 1

while c <= 10:
  print(f'{t} - ', end=' ')
  t += razao
  c += 1

PROGRESSÃO ARITIMÉTICA
Primeiro Termo:2
Razão:2
2 -  4 -  6 -  8 -  10 -  12 -  14 -  16 -  18 -  20 -  

Exercício Python 62: Melhore o DESAFIO 61, perguntando para o usuário se ele quer mostrar mais alguns termos. O programa encerrará quando ele disser que quer mostrar 0 termos.

In [ ]:
print ('PROGRESSÃO ARITIMÉTICA')

termo = int(input('Primeiro Termo:'))
razao = int(input('Razão:'))
t = termo
c = 1
total = 0
mais_termos = 10
while mais_termos != 0:
  total = total + mais_termos
  while c <= total:
    print(f'{t} -> ', end=' ')
    t += razao
    c += 1
  mais_termos = int(input('Quantos termos a mais você quer mostrar?'))
print('FIM')


PROGRESSÃO ARITIMÉTICA
Primeiro Termo:5
Razão:2
5 ->  7 ->  9 ->  11 ->  13 ->  15 ->  17 ->  19 ->  21 ->  23 ->  Quantos termos a mais você quer mostrar?5
25 ->  27 ->  29 ->  31 ->  33 ->  Quantos termos a mais você quer mostrar?2
35 ->  37 ->  Quantos termos a mais você quer mostrar?0
FIM


Exercício Python 63: Escreva um programa que leia um número N inteiro qualquer e mostre na tela os N primeiros elementos de uma Sequência de Fibonacci. Exemplo:

0 – 1 – 1 – 2 – 3 – 5 – 8

In [ ]:

n = int(input("Digite o número de elementos da sequência de Fibonacci: ")) #mostra o número que o usuário quer mostrar da sequência
fibonacci = [0, 1]
for i in range(2, n):
    fibonacci.append(fibonacci[i - 1] + fibonacci[i - 2])
print("Sequência de Fibonacci com os primeiros", n, "elementos:")
for num in fibonacci:
    print(num, end=" ")


Digite o número de elementos da sequência de Fibonacci: 5
Sequência de Fibonacci com os primeiros 5 elementos:
0 1 1 2 3 

In [ ]:
n = int(input("Digite o número de elementos da sequência de Fibonacci: "))

fibonacci = [0, 1]  # Inicializando a sequência de Fibonacci com os dois primeiros elementos
i = 2  # Contador para os próximos elementos da sequência

# Gerando os próximos elementos da sequência de Fibonacci até alcançar n elementos
while i < n:
    next_fib = fibonacci[-1] + fibonacci[-2]  # Calculando o próximo número da sequência
    fibonacci.append(next_fib)  # Adicionando o próximo número à sequência
    i += 1  # Incrementando o contador

# Mostrando os primeiros n elementos da sequência de Fibonacci
print("Sequência de Fibonacci com os primeiros", n, "elementos:")
for num in fibonacci:
    print(num, end=" ")


Digite o número de elementos da sequência de Fibonacci: 5
Sequência de Fibonacci com os primeiros 5 elementos:
0 1 1 2 3 

Exercício Python 64: Crie um programa que leia vários números inteiros pelo teclado. O programa só vai parar quando o usuário digitar o valor 999, que é a condição de parada. No final, mostre quantos números foram digitados e qual foi a soma entre eles (desconsiderando o flag).

In [ ]:
# Inicializando variáveis
soma = contador = 0

# Loop para ler os números e somar
while True:
    num = int(input("Digite um número inteiro (999 para parar): "))

    # Verifica se o número é 999 para parar o loop
    if num == 999:
        break

    soma += num  # Adiciona o número à soma
    contador += 1  # Incrementa o contador de números digitados

# Exibindo o resultado
print(f"Foram digitados {contador} números.")
print(f"A soma dos números digitados é {soma}.")


Exercício Python 65: Crie um programa que leia vários números inteiros pelo teclado. No final da execução, mostre a média entre todos os valores e qual foi o maior e o menor valores lidos. O programa deve perguntar ao usuário se ele quer ou não continuar a digitar valores.

In [ ]:
  # Inicializando variáveis
soma = contador = 0
maior = menor = 0

while True:
    num = int(input("Digite um número inteiro: "))

    # Atualizando soma e contador
    soma += num
    contador += 1

    # Atualizando maior e menor
    if contador == 1:
        maior = menor = num
    else:
        if num > maior:
            maior = num
        if num < menor:
            menor = num

    continuar = input("Deseja continuar? (S/N): ").strip().upper()
    if continuar != 'S':
        break

# Calculando a média
media = soma / contador

# Exibindo os resultados
print(f"Média dos valores digitados: {media}")
print(f"Maior valor digitado: {maior}")
print(f"Menor valor digitado: {menor}")


Exercício Python 66: Crie um programa que leia números inteiros pelo teclado. O programa só vai parar quando o usuário digitar o valor 999, que é a condição de parada. No final, mostre quantos números foram digitados e qual foi a soma entre elas (desconsiderando o flag).

Exercício Python 67: Faça um programa que mostre a tabuada de vários números, um de cada vez, para cada valor digitado pelo usuário. O programa será interrompido quando o número solicitado for negativo.

Exercício Python 68: Faça um programa que jogue par ou ímpar com o computador. O jogo só será interrompido quando o jogador perder, mostrando o total de vitórias consecutivas que ele conquistou no final do jogo.

Exercício Python 69: Crie um programa que leia a idade e o sexo de várias pessoas. A cada pessoa cadastrada, o programa deverá perguntar se o usuário quer ou não continuar. No final, mostre:

A) quantas pessoas tem mais de 18 anos.

B) quantos homens foram cadastrados.

C) quantas mulheres tem menos de 20 anos.

In [ ]:
from time import sleep

print('CADASTRO DE USUÁRIOS')
print('*'*21)
print('INSTRUÇÕES\n Idade: Digite um número inteiro. \n Sexo: Feminino (F), Masculino (M), Outro (O).\n')
print('*'*21)
dados_masculinos = dados_femininos = dados_idade = dados_outro = 0

while True:
  idade = int(input('Idade: '))
  sexo = ' '
  while sexo not in 'FMO':
    sexo = input('Sexo: (F/M/O) ').strip().upper()[0]
  if idade > 18:
    dados_idade +=1
  if sexo == 'M':
    dados_masculinos +=1
  if sexo == 'F' and idade < 20:
    dados_femininos += 1
  if sexo =='O':
    dados_outro += 1
  resposta = ' '
  while resposta not in 'SN':
    resposta = input('Quer continuar? S/N ').strip().upper()[0]
  if resposta == 'N':
    break
sleep(1)
print('PROCESSANDO... ', end=' ')
sleep(3)
print('Dados Cadastrados com Sucesso.'.upper())
print(f'Total de pessoas com mais de 18 anos: {dados_idade}.')
print(f'Total de homens cadastrados: {dados_masculinos}.')
print (f'Total de mulheres com menos de 20 anos: {dados_femininos }.')
print (f'Total de pessoas identificadas com gênero neutro. {dados_outro }.')

CADASTRO DE USUÁRIOS
*********************
INSTRUÇÕES
 Idade: Digite um número inteiro. 
 Sexo: Feminino (F), Masculino (M), Outro (O).

*********************
Idade: 22
Sexo: (F/M/O) f
Quer continuar? S/N s
Idade: 17
Sexo: (F/M/O) m
Quer continuar? S/N s
Idade: 19
Sexo: (F/M/O) o
Quer continuar? S/N s
Idade: 14
Sexo: (F/M/O) f
Quer continuar? S/N s
Idade: 77
Sexo: (F/M/O) m
Quer continuar? S/N s
Idade: 55
Sexo: (F/M/O) o
Quer continuar? S/N n
PROCESSANDO...  DADOS CADASTRADOS COM SUCESSO.
Total de pessoas com mais de 18 anos: 4
Total de homens cadastrados: 2
Total de mulheres com menos de 20 anos: 1 mulheres tem menos de 20 anos.
Total de pessoas identificadas com gênero neutro. 2.


Exercício Python 70: Crie um programa que leia o nome e o preço de vários produtos. O programa deverá perguntar se o usuário vai continuar ou não. No final, mostre:

A) qual é o total gasto na compra.

B) quantos produtos custam mais de R$1000.

C) qual é o nome do produto mais barato.

In [ ]:
total = produto_maior_1000 = produto_barato = c = 0
nome_barato = ' '

while True:
  nome_produto = input('Digite o nome do produto: ').strip().upper()
  preco_produto = float(input('Digite o preço do produto: R$ '))
  total += preco_produto
  c +=1

  if preco_produto > 1000:
    produto_maior_1000 += 1

  if c == 1:
    produto_barato = preco_produto
    nome_barato = nome_produto

  else:
    if preco_produto < produto_barato:
      produto_barato = preco_produto
      nome_barato = nome_produto

  loop = ' '
  while loop not in 'SN':
    loop = input('Vai continuar? [S/N] ').strip().upper()[0]
  if loop == 'N':
    break
print(total)
print(produto_maior_1000)
print(nome_barato)
print('FIM.')

# A) qual é o total gasto na compra.
# B) quantos produtos custam mais de R$1000.
# C) qual é o nome do produto mais barato.

Digite o nome do produto: Computador
Digite o preço do produto: R$ 5000
Vai continuar? [S/N] s
Digite o nome do produto: Leite
Digite o preço do produto: R$ 2
Vai continuar? [S/N] s
Digite o nome do produto: Carro 
Digite o preço do produto: R$ 50000
Vai continuar? [S/N] n
55002.0
2
LEITE
FIM.


Reforço para a verificação do menor número.

In [ ]:
#checar o produto mais barato
produto_barato = c = 0

while c <5:
  preco_produto = float(input('Digite o preço do produto: R$ '))
  c +=1

  if c == 1:
    produto_barato = preco_produto

  else:
    if preco_produto < produto_barato:
      produto_barato = preco_produto

print(f'R$ {produto_barato}')
print('FIM.')



Digite o preço do produto: R$ 4
Digite o preço do produto: R$ 5
Digite o preço do produto: R$ 2
Digite o preço do produto: R$ 7
Digite o preço do produto: R$ 4
R$ 2.0
FIM.


Exercício Python 071: Crie um programa que simule o funcionamento de um caixa eletrônico. No início, pergunte ao usuário qual será o valor a ser sacado (número inteiro) e o programa vai informar quantas cédulas de cada valor serão entregues. OBS:

considere que o caixa possui cédulas de R$50, R$20, R$10 e R$1.

In [ ]:
saque = int(input('Qual o valor que deseja sacar? R$ '))



In [ ]:
print('=' * 30)
print('{:^30}'.format('BANCO CEV'))
print('=' * 30)

valor_saque = int(input('Qual valor você deseja sacar? R$ '))

total = valor_saque
cedula = 50
total_cedulas = 0

while True:
    if total >= cedula:
        total -= cedula
        total_cedulas += 1
    else:
        if total_cedulas > 0:
            print(f'Total de {total_cedulas} cédulas de R$ {cedula}')
        if cedula == 50:
            cedula = 20
        elif cedula == 20:
            cedula = 10
        elif cedula == 10:
            cedula = 1
        total_cedulas = 0
        if total == 0:
            break

print('=' * 30)
print('Volte sempre ao BANCO CEV! Tenha um bom dia!')


          BANCO CEV           
Qual valor você deseja sacar? R$ 100
Total de 2 cédulas de R$ 50
Volte sempre ao BANCO CEV! Tenha um bom dia!


Exercício Python 72: Crie um programa que tenha uma dupla totalmente preenchida com uma contagem por extenso, de zero até vinte. Seu programa deverá ler um número pelo teclado (entre 0 e 20) e mostrá-lo por extenso.

In [ ]:
extenso = ('zero',' um', 'dois',' três', 'quatro', 'cinco',
           'seis',' sete',' oito', 'nove', 'dez', 'onze',
           'doze', 'treze', 'quatorze', 'quinze', 'dezesseis',
           'dezesete', 'dezoito', 'dezenove', 'vinte')

while True:
  num = int(input('Digite um número inteiro de 1 a 20: '))
  if 0 <= num <= 20:
    break
  print('Tente novamente.', end=' ')
print(f'Você digitou o número {extenso[num]}')

Digite um número inteiro de 1 a 20: 44
Tente novamente. Digite um número inteiro de 1 a 20: 44
Tente novamente. Digite um número inteiro de 1 a 20: 33
Tente novamente. Digite um número inteiro de 1 a 20: 23
Tente novamente. Digite um número inteiro de 1 a 20: 12
Você digitou o número doze


Exercício Python 73: Crie uma tupla preenchida com os 20 primeiros colocados da Tabela do Campeonato Brasileiro de Futebol, na ordem de colocação. Depois mostre:

a) Os 5 primeiros times.

b) Os últimos 4 colocados.

c) Times em ordem alfabética.

d) Em que posição está o time da Fluminense.

In [ ]:
times =  ('Palmeiras', 'Grêmio', 'Atlético-MG', 'Flamengo', 'Botafogo',
       'Bragantino', 'Fluminense', 'Athletico-PR', 'Internacional',
       'Fortaleza', 'São Paulo', 'Cuiabá', 'Corinthians', 'Cruzeiro',
       'Vasco', 'Bahia', 'Santos', 'Goiás', 'Coritiba', 'América-MG')

times_ordenados = sorted(times)
posicao_flu = times.index('Fluminense')

print('@'*30)
print('TABELA DO BRASILEIRÃO'.center(30))
print('@'*30)


print('\nLISTA DE TIMES')
print('=*'*15)
for t in times:
  print(t)

print('\nTOP 5')
print('=*'*15)
for t in times[0:5]:
  print(t)

print('\nNA LANTERNA')
print('=*'*15)
for l in times[-4:]:
  print(l)

print('\nORDEN ALFABÉTICA')
print('=*'*15)
for t_o in times_ordenados:
  print(t_o)

print('\nPOSIÇÃO FLUMINENSE')
print('=*'*15)
print(f'O Fluminense está na {posicao_flu}ª posição.')


@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
    TABELA DO BRASILEIRÃO     
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

LISTA DE TIMES
=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*
Palmeiras
Grêmio
Atlético-MG
Flamengo
Botafogo
Bragantino
Fluminense
Athletico-PR
Internacional
Fortaleza
São Paulo
Cuiabá
Corinthians
Cruzeiro
Vasco
Bahia
Santos
Goiás
Coritiba
América-MG

TOP 5
=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*
Palmeiras
Grêmio
Atlético-MG
Flamengo
Botafogo

NA LANTERNA
Santos
Goiás
Coritiba
América-MG

ORDEN ALFABÉTICA
América-MG
Athletico-PR
Atlético-MG
Bahia
Botafogo
Bragantino
Corinthians
Coritiba
Cruzeiro
Cuiabá
Flamengo
Fluminense
Fortaleza
Goiás
Grêmio
Internacional
Palmeiras
Santos
São Paulo
Vasco

POSIÇÃO FLUMINENSE
O Fluminense está na 6ª posição.


Exercício Python 074: Crie um programa que vai gerar cinco números aleatórios e colocar em uma tupla. Depois disso, mostre a listagem de números gerados e também indique o menor e o maior valor que estão na tupla.

In [ ]:
from random import randint
num_aleatorios = (randint(1,100), randint(1,100), randint(1,100), randint(1,100), randint(1,100))
print('Os números sorteados foram: ')
for n in num_aleatorios:
    print(n, end=' ')
print(f'\nO maior número foi o {max(num_aleatorios)}')
print(f'O menor número foi o {min(num_aleatorios)}')


Os números sorteados foram: 
64 15 42 79 34 
O maior número foi o 79
O menor número foi o 15


Exercício Python 075: Desenvolva um programa que leia quatro valores pelo teclado e guarde-os em uma tupla. No final, mostre:

A) Quantas vezes apareceu o valor 9.

B) Em que posição foi digitado o primeiro valor 3.

C) Quais foram os números pares.

In [ ]:
# Solicitar ao usuário que insira 4 números inteiros
numeros = tuple(int(input(f"Digite um número inteiro: ")) for i in range(4))

print("Os números digitados foram:", numeros)

print(f'O número 9 apareceu {numeros.count(9)} vezes.')

if 3 in numeros:
  print(f'O número 3 aparece na posição de índece {numeros.index(3)+1}')
else:
  print('O valor 3 não foi digitado.')

print('O valores pares são: ', end='')
for n in numeros:
  if n % 2 ==0:
    print(n, end=' ')

Digite um número inteiro: 2
Digite um número inteiro: 3
Digite um número inteiro: 4
Digite um número inteiro: 5
Os números digitados foram: (2, 3, 4, 5)
O número 9 apareceu 0 vezes.
O número 3 aparece na posição de índece 2
O valores pares são: 2, 4, 

Exercício Python 076: Crie um programa que tenha uma tupla única com nomes de produtos e seus respectivos preços, na sequência. No final, mostre uma listagem de preços, organizando os dados em forma tabular.

In [ ]:
# Tupla de produtos e preços
produtos_precos = (
    ("Arroz", 10.99),
    ("Feijão", 6.49),
    ("Macarrão", 3.99),
    ("Café", 8.99),
    ("Leite", 4.59)
)

# Imprimir cabeçalho da tabela
print("=" * 30)
print("{:<15} {:>10}".format("Produto", "Preço"))
print("=" * 30)

# Imprimir os produtos e preços em forma tabular
for produto, preco in produtos_precos:
    print("{:<15} R${:>8.2f}".format(produto, preco))

# Imprimir linha final da tabela
print("=" * 30)


Produto              Preço
Arroz           R$   10.99
Feijão          R$    6.49
Macarrão        R$    3.99
Café            R$    8.99
Leite           R$    4.59


Exercício Python 077: Crie um programa que tenha uma tupla com várias palavras (não usar acentos). Depois disso, você deve mostrar, para cada palavra, quais são as suas vogais.

In [ ]:
palavras = ('aprender', 'trabalhar', 'programar', 'curso', 'python', 'online', 'trabalho', 'intuitivo')

for p in palavras:
  print(f'\nPara a paralva {p.upper()}, temos as vogais: ', end='')
  for vogal in p:
    if vogal in 'aeiou':
      print(vogal, end=' ')


Para a paralva APRENDER, temos as vogais: a e e 
Para a paralva TRABALHAR, temos as vogais: a a a 
Para a paralva PROGRAMAR, temos as vogais: o a a 
Para a paralva CURSO, temos as vogais: u o 
Para a paralva PYTHON, temos as vogais: o 
Para a paralva ONLINE, temos as vogais: o i e 
Para a paralva TRABALHO, temos as vogais: a a o 
Para a paralva INTUITIVO, temos as vogais: i u i i o 

Esses são métodos usados em listas em Python para adicionar, ordenar, inserir e remover elementos. Aqui está uma explicação de cada um deles:

1. **.append():**
   - O método `.append()` é usado para adicionar um elemento ao final de uma lista.
   - Sintaxe: `lista.append(elemento)`
   - Exemplo:
     ```python
     lista = [1, 2, 3]
     lista.append(4)
     print(lista)  # Saída: [1, 2, 3, 4]
     ```

2. **.sort():**
   - O método `.sort()` é usado para ordenar os elementos de uma lista em ordem crescente ou alfabética.
   - Para ordenar em ordem decrescente, você pode usar o argumento `reverse=True`.
   - Sintaxe: `lista.sort(reverse=False)`
   - Exemplo:
     ```python
     lista = [3, 1, 4, 2]
     lista.sort()
     print(lista)  # Saída: [1, 2, 3, 4]
     ```

3. **.insert():**
   - O método `.insert()` é usado para inserir um elemento em uma posição específica da lista.
   - Sintaxe: `lista.insert(posição, elemento)`
   - Exemplo:
     ```python
     lista = [1, 2, 3]
     lista.insert(1, 4)
     print(lista)  # Saída: [1, 4, 2, 3]
     ```

4. **.pop():**
   - O método `.pop()` é usado para remover e retornar o último elemento de uma lista. Também pode ser usado para remover um elemento em uma posição específica.
   - Sintaxe: `lista.pop([posição])`
   - Exemplo:
     ```python
     lista = [1, 2, 3]
     elemento = lista.pop()
     print(elemento)  # Saída: 3
     print(lista)  # Saída: [1, 2]
     ```
   - Se uma posição específica for fornecida, `.pop()` removerá e retornará o elemento nessa posição.

Esses são métodos muito úteis para manipular listas em Python, permitindo adicionar, ordenar, inserir e remover elementos de forma eficiente.

Exercício Python 078: Faça um programa que leia 5 valores numéricos e guarde-os em uma lista. No final, mostre qual foi o maior e o menor valor digitado e as suas respectivas posições na lista.

In [ ]:
# Cria uma lista vazia para armazenar os valores
valores = []

# Lê os 5 valores numéricos
for i in range(5):
    valor = float(input(f'Digite o {i+1}º valor: '))
    valores.append(valor)

# Encontra o maior e o menor valor na lista, juntamente com suas posições
maior_valor = max(valores)
menor_valor = min(valores)
posicao_maior = valores.index(maior_valor) + 1
posicao_menor = valores.index(menor_valor) + 1

# Exibe os resultados
print(f'O maior valor digitado foi {maior_valor} na posição {posicao_maior}')
print(f'O menor valor digitado foi {menor_valor} na posição {posicao_menor}')


Digite o 1º valor: 2
Digite o 2º valor: 2
Digite o 3º valor: 1
Digite o 4º valor: 3
Digite o 5º valor: 5
O maior valor digitado foi 5.0 na posição 5
O menor valor digitado foi 1.0 na posição 3


In [ ]:
#colocar valores em uma lista
lista = []

for d in range(5):
  items = int(input('Digite um valor: '))
  lista.append(items)
print(lista)


Digite um valor: 1
Digite um valor: 2
Digite um valor: 3
Digite um valor: 4
Digite um valor: 5
[1, 2, 3, 4, 5]


In [ ]:
lista_nomes = []

for d in range(5):
  nomes = input('Digite um nome: ')
  lista_nomes.append(nomes)

lista_nomes.sort()

print(lista_nomes)

Digite um nome: Clara
Digite um nome: Gabriel
Digite um nome: Raquel
Digite um nome: Maria
Digite um nome: Sarah
['Clara', 'Gabriel', 'Maria', 'Raquel', 'Sarah']


In [ ]:
lista_numeros = []

for d in range(5):
  lista = int(input('Digite um número: '))
  lista_numeros.append(lista)

lista_numeros.sort()

print(f'O menor item da lista é {lista_numeros[0]}')
print(f'O maior item da lista é {lista_numeros[-1]}')

Digite um número: 3
Digite um número: 4
Digite um número: 5
Digite um número: 2
Digite um número: 7
O menor item da lista é 2
O maior item da lista é 7


Exercício Python 079: Crie um programa onde o usuário possa digitar vários valores numéricos e cadastre-os em uma lista. Caso o número já exista lá dentro, ele não será adicionado. No final, serão exibidos todos os valores únicos digitados, em ordem crescente.



Exercício Python 080: Crie um programa onde o usuário possa digitar cinco valores numéricos e cadastre-os em uma lista, já na posição correta de inserção (sem usar o sort()). No final, mostre a lista ordenada na tela.

Exercício Python 081: Crie um programa que vai ler vários números e colocar em uma lista.                  Depois disso, mostre:                                                                                                                                                A) Quantos números foram digitados.                                                                                                                    B) A lista de valores, ordenada de forma decrescente.                                                                                          C) Se o valor 5 foi digitado e está ou não na lista.

Exercício Python 082: Crie um programa que vai ler vários números e colocar em uma lista. Depois disso, crie duas listas extras que vão conter apenas os valores pares e os valores ímpares digitados, respectivamente. Ao final, mostre o conteúdo das três listas geradas.

Exercício Python 083: Crie um programa onde o usuário digite uma expressão qualquer que use parênteses. Seu aplicativo deverá analisar se a expressão passada está com os parênteses abertos e fechados na ordem correta.

In [ ]:
dados = []

for i in range(3):
  resposta = input('Digite um dado: ')
  dados.append(resposta)
print(dados)



Digite um dado: Dublin
Digite um numero: 25
Digite um nome: Michel
Digite um dado: Belfast
Digite um numero: 33
Digite um nome: Carlos
Digite um dado: Cavan
Digite um numero: 44
Digite um nome: Daniela
['Michel', 'Carlos', 'Daniela']


In [ ]:
teste = []
teste.append('Gustavo')
teste.append('40')
galera = []
galera.append(teste[:]) #cria uma copia
teste[0] = 'João'
teste [1] = 22
galera.append(teste[:])
print(galera)
print(type(galera))



[['Gustavo', '40'], ['João', 22]]
<class 'list'>


In [ ]:
galera = [['Maria', 19], ['João', 33], ['Joaquim', 14], ['Ana', 41]]
soma_idades = 0
nomes = []

for p in galera:
  print(f'{p[0]} tem {p[1]} anos de idade.')
  soma_idades += p[1]
  nomes.append(p[0])

media = soma_idades/len(galera)
nomes.sort()
print(nomes)
print(f'A média de idade da galera eh {media} anos de idade.')


Maria tem 19 anos de idade.
João tem 33 anos de idade.
Joaquim tem 14 anos de idade.
Ana tem 41 anos de idade.
['Ana', 'Joaquim', 'João', 'Maria']
A média de idade da galera eh 26.75 anos de idade.


In [ ]:
galera = []
dado = []
maior = menor = 0
for d in range(3):
  dado.append(input('NOME: '))
  dado.append(int(input('IDADE: ')))
  galera.append(dado[:]) #copia do dado
  dado.clear()

for p in galera:
  if p[1] > 18:
    print(f'{p[0]} eh maior de idade.')
    maior += 1
  else:
    print(f'{p[0]} eh menor de idade.')
    menor +=1

print(f'Temos {maior} maiores e {menor} menores de idade')

NOME: Clara
IDADE: 12
NOME: Gabriel
IDADE: 9
NOME: Sarah
IDADE: 22
Clara eh menor de idade.
Gabriel eh menor de idade.
Sarah eh maior de idade.
Temos 1 maiores e 2 menores de idade


Exercício Python 084: Faça um programa que leia nome e peso de várias pessoas,                                      guardando tudo em uma lista. No final, mostre:                                                                                                   
 A) Quantas pessoas foram cadastradas.
 B) Uma listagem com as pessoas mais pesadas.
 C) Uma listagem com as pessoas mais leves.

Exercício Python 085: Crie um programa onde o usuário possa digitar sete valores numéricos e cadastre-os em uma lista única que mantenha separados os valores pares e ímpares. No final, mostre os valores pares e ímpares em ordem crescente.

Exercício Python 086: Crie um programa que declare uma matriz de dimensão 3×3 e preencha com valores lidos pelo teclado. No final, mostre a matriz na tela, com a formatação correta.

Exercício Python 087: Aprimore o desafio anterior, mostrando no final:                                                    A) A soma de todos os valores pares digitados.                                                                                                  B) A soma dos valores da terceira coluna.                                                                                                                C) O maior valor da segunda linha.

Exercício Python 087: Aprimore o desafio anterior, mostrando no final:                                                    A) A soma de todos os valores pares digitados.                                                                                                  B) A soma dos valores da terceira coluna.                                                                                                                C) O maior valor da segunda linha.

Exercício Python 088: Faça um programa que ajude um jogador da MEGA SENA a criar palpites.O programa vai perguntar quantos jogos serão gerados e vai sortear 6 números entre 1 e 60 para cada jogo, cadastrando tudo em uma lista composta.

Exercício Python 089: Crie um programa que leia nome e duas notas de vários alunos e guarde tudo em uma lista composta. No final, mostre um boletim contendo a média de cada um e permita que o usuário possa mostrar as notas de cada aluno individualmente.

Aula de Dic

In [ ]:
pessoas = {'nome':'Gustavo', 'sexo':'M','idade':22}

print(f'{pessoas["nome"]} tem {pessoas["idade"]} anos de idade.')
print(pessoas.keys())
print(pessoas.items())
print(pessoas.values())

for k in pessoas.keys():
  print(k)

for k in pessoas.values():
  print(k)

for k, v in pessoas.items(): #chave e valor
  print(k, v)

del pessoas["sexo"]
for k, v in pessoas.items(): #chave e valor
  print(k, v)

pessoas['peso'] = 98.5 #adiciona sem usar o .append
for k, v in pessoas.items(): #chave e valor
  print(k, v)

Gustavo tem 22 anos de idade.
dict_keys(['nome', 'sexo', 'idade'])
dict_items([('nome', 'Gustavo'), ('sexo', 'M'), ('idade', 22)])
dict_values(['Gustavo', 'M', 22])
nome
sexo
idade
Gustavo
M
22
nome Gustavo
sexo M
idade 22
nome Gustavo
idade 22
nome Gustavo
idade 22
peso 98.5


In [ ]:
brasil = []
estado_1 = {'estado': 'Rio de Janeiro', 'uf': 'RJ'}
estado_2 = {'estado': 'São Paulo', 'uf': 'SP'}
brasil.append(estado_1)
brasil.append(estado_2)

print(brasil) #Uma lista com 2 dicionários
print(brasil[0]['uf'])



[{'estado': 'Rio de Janeiro', 'uf': 'RJ'}, {'estado': 'São Paulo', 'uf': 'SP'}]
RJ


In [ ]:
estados = {} #cria um dic
brasil = [] #cria uma lista

for c in range(2):
  estados['uf'] = input('ESTADO: ')
  estados['sigla'] = input('SIGLA: ')
  brasil.append(estados.copy()) #dic não funciona com append
for e in brasil:
  for k, v, in e.items():
    print(v)


KeyboardInterrupt: Interrupted by user

Exercício Python 090: Faça um programa que leia nome e média de um aluno, guardando também a situação em um dicionário. No final, mostre o conteúdo da estrutura na tela.

In [ ]:
#construindo um dicionário com dados de input
aluno = {}
aluno['nome'] = input('NOME DO ALUNO: ')
aluno['media'] = float(input(f'MÉDIA DE {aluno["nome"]}: ')) #declarar com aspas duplas quando estiver dentro de uma string com aspas simples.
if aluno['media'] >= 7:
  aluno['situacao'] = 'APROVADO'
elif aluno['media'] < 7 and aluno['media'] >= 5:
  aluno['situacao'] = 'RECUPERAÇÃO'
else:
  aluno['situacao'] = 'REPROVADO'

for k, v in aluno.items():
  print(f'{k} é igual a {v}')


NOME DO ALUNO: Clara
MÉDIA DE Clara: 8
nome é igual a Clara
media é igual a 8.0
situacao é igual a APROVADO


Exercício Python 091: Crie um programa onde 4 jogadores joguem um dado e tenham resultados aleatórios. Guarde esses resultados em um dicionário em Python. No final, coloque esse dicionário em ordem, sabendo que o vencedor tirou o maior número no dado.

In [ ]:
from random import randint
from time import sleep
from operator import itemgetter

jogo = {'jogador1': randint(1,6), 'jorgador2': randint(1,6), 'jorgador3': randint(1,6), 'jorgador4': randint(1,6)}
ranking = () #lista fica mais fácil fazer o ranking
for k, v in jogo.items():
  print(f'O {k} rolou o dado e o resultado foi {v}!')
  sleep(2)
ranking = sorted(jogo.items(), key=itemgetter(1), reverse=True)
print('=*'*30)
print(' == RANKING DOS JOGADORES == ')
print('=*'*30)

for i, v in enumerate(ranking):
  print(f'{i+1}º lugar: {v[0]} tirou {v[1]}')


O jogador1 rolou o dado e o resultado foi 3!
O jorgador2 rolou o dado e o resultado foi 1!
O jorgador3 rolou o dado e o resultado foi 3!
O jorgador4 rolou o dado e o resultado foi 1!
=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*
 == RANKING DOS JOGADORES == 
=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*=*
1º lugar: jogador1 tirou 3
2º lugar: jorgador3 tirou 3
3º lugar: jorgador2 tirou 1
4º lugar: jorgador4 tirou 1


Exercício Python 092: Crie um programa que leia nome, ano de nascimento e carteira de trabalho e cadastre-o (com idade) em um dicionário. Se por acaso a CTPS for diferente de ZERO, o dicionário receberá também o ano de contratação e o salário. Calcule e acrescente, além da idade, com quantos anos a pessoa vai se aposentar.

In [ ]:
#Para esse exercícios, vamos criar um dicionário chamado dados
from datetime import datetime
dados = {}
dados['nome'] = input('Nome: ')
ano_nasc = int(input('Ano de Nascimento: '))
dados['idade'] = datetime.now().year - ano_nasc
dados['ctps'] = int(input('CPTS [Digite 0 caso não tenha carteira de trabalho]:  '))
if dados['ctps'] != 0:
  dados['contratação'] = int(input('Ano de contratação: '))
  dados['salario'] = int(input('Salário: '))
#Para a aposentadoria, vamos calcular o ano que ele vai se aposentar.
#Isso seria o ano da contratação + ano de contribuição (que para esse exercicio serão 35 anos). Isso resulta no ano da aposentodira.
#Exeplo ano de contratação = 2000 + 35. O ano de aposentadoria é 2035
#Mas o exercicios pede com quantos anos a pessoa vai de aposentar. Temos que descobrir quantos anos a pessoa terá no ano da aposentadoria.
#Para isso, vamos diminuir o ano de aposentadoria - o ano atual. E depois somar a idade atual da pessoa.
  dados['aposentadoria'] = dados['idade'] + ((dados['contratação'] + 35) - datetime.now().year)
for k, v in dados.items():
  print(f'- {k} = {v}'.upper())



Nome: Clara Mendes
Ano de Nascimento: 1982
CPTS [Digite 0 caso não tenha carteira de trabalho]:  58875
Ano de contratação: 2006
Salário: 3000
- NOME = CLARA MENDES
- IDADE = 42
- CTPS = 58875
- CONTRATAÇÃO = 2006
- SALARIO = 3000
- APOSENTADORIA = 59


Exercício Python 093: Crie um programa que gerencie o aproveitamento de um jogador de futebol. O programa vai ler o nome do jogador e quantas partidas ele jogou. Depois vai ler a quantidade de gols feitos em cada partida. No final, tudo isso será guardado em um dicionário, incluindo o total de gols feitos durante o campeonato.

In [ ]:
dados = {}
dados['nome'] = input('NOME DO JOGADOR: ')
partidas = int(input( 'NÚMERO DE PARTIDAS: '))
gol = []
for p in range(partidas):
  gol.append(int(input(f'NÚMERO DE GOLS NA {p+1}ª PARTIDA: '))) #isso aqui é uma lista
dados['gols'] = gol[:] #[:] para copiar fatiado
dados['total'] = sum(gol)
for k, v in dados.items(): #para ler o dicionário
  print(f'{k} = {v}'.upper())
for i, v in enumerate(dados['gols']): #ler a lista dentro do dicionário
  print(f'Na partida {1} fez {v} gols')

print(f'O jogador {dados["nome"]} fez um total de {dados["total"]} gols.')


NOME DO JOGADOR: Zico
NÚMERO DE PARTIDAS: 1
NÚMERO DE GOLS NA 1ª PARTIDA: 2
NOME = ZICO
GOLS = [2]
TOTAL = 2
Na partida 1 fez 2 gols
O jogador Zico fez um total de 2 gols.


Exercício Python 094: Crie um programa que leia nome, sexo e idade de várias pessoas, guardando os dados de cada pessoa em um dicionário e todos os dicionários em uma lista. No final, mostre: A) Quantas pessoas foram cadastradas B) A média de idade C) Uma lista com as mulheres D) Uma lista de pessoas com idade acima da média

In [ ]:
dados = {}
dados['nome'] = input('NOME: ')
dados['sexo'] = input('SEXO: [F/M/O]').strip().upper()
dados['idade'] = int(input('IDADE: '))

print(dados)
#cadastros
#idade_media
#lista_mulheres
#pessoas_acima_idade

NOME: Clara
SEXO: [F/M/O]cF
IDADE: 41
{'nome': 'Clara', 'sexo': 'F', 'idade': 41}


Exercício Python 095: Aprimore o desafio 93 para que ele funcione com vários jogadores, incluindo um sistema de visualização de detalhes do aproveitamento de cada jogador.

Exercício Python 096: Faça um programa que tenha uma função chamada área(), que receba as dimensões de um terreno retangular (largura e comprimento) e mostre a área do terreno.


Exercício Python 097: Faça um programa que tenha uma função chamada escreva(), que receba um texto qualquer como parâmetro e mostre uma mensagem com tamanho adaptável.                                  Ex:                                                                                                                                                                        escreva(‘Olá, Mundo!’) Saída:                                                                                                                          ~~~~~~~~~                                                                                                                                                            Olá, Mundo!                                                            

Exercício Python 098: Faça um programa que tenha uma função chamada contador(), que receba três parâmetros: início, fim e passo. Seu programa tem que realizar três contagens através da função criada:                                                                                                                                                                            a) de 1 até 10, de 1 em 1                                                                                                                                              b) de 10 até 0, de 2 em 2                                                                                                                                            c) uma contagem personalizada

Exercício Python 099: Faça um programa que tenha uma função chamada maior(), que receba vários parâmetros com valores inteiros. Seu programa tem que analisar todos os valores e dizer qual deles é o maior.

Exercício Python 100: Faça um programa que tenha uma lista chamada números e duas funções chamadas sorteia() e somaPar(). A primeira função vai sortear 5 números e vai colocá-los dentro da lista e a segunda função vai mostrar a soma entre todos os valores pares sorteados pela função anterior.

Exercício Python 101: Crie um programa que tenha uma função chamada voto() que vai receber como parâmetro o ano de nascimento de uma pessoa, retornando um valor literal indicando se uma pessoa tem voto NEGADO, OPCIONAL e OBRIGATÓRIO nas eleições.

Exercício Python 102: Crie um programa que tenha uma função fatorial() que receba dois parâmetros: o primeiro que indique o número a calcular e outro chamado show, que será um valor lógico (opcional) indicando se será mostrado ou não na tela o processo de cálculo do fatorial.

Exercício Python 103: Faça um programa que tenha uma função chamada ficha(), que receba dois parâmetros opcionais: o nome de um jogador e quantos gols ele marcou. O programa deverá ser capaz de mostrar a ficha do jogador, mesmo que algum dado não tenha sido informado corretamente.

Exercício Python 104: Crie um programa que tenha a função leiaInt(), que vai funcionar de forma semelhante ‘a função input() do Python, só que fazendo a validação para aceitar apenas um valor numérico. Ex: n = leiaInt(‘Digite um n: ‘)

xercício Python 105: Faça um programa que tenha uma função notas() que pode receber várias notas de alunos e vai retornar um dicionário com as seguintes informações:

– Quantidade de notas                                                                                                                                                  – A maior nota                                                                                                                                                                – A menor nota                                                                                                                                                              – A média da turma                                                                                                                                                      – A situação (opcional)

Exercício Python 106: Faça um mini-sistema que utilize o Interactive Help do Python. O usuário vai digitar o comando e o manual vai aparecer. Quando o usuário digitar a palavra ‘FIM’, o programa se encerrará. Importante: use cores.


Exercício Python 107: Crie um módulo chamado moeda.py que tenha as funções incorporadas aumentar(), diminuir(), dobro() e metade(). Faça também um programa que importe esse módulo e use algumas dessas funções.

Exercício Python 108: Adapte o código do desafio #107, criando uma função adicional chamada moeda() que consiga mostrar os números como um valor monetário formatado.

Exercício Python 109: Modifique as funções que form criadas no desafio 107 para que elas aceitem um parâmetro a mais, informando se o valor retornado por elas vai ser ou não formatado pela função moeda(), desenvolvida no desafio 108.

Exercício Python 110: Adicione o módulo moeda.py criado nos desafios anteriores, uma função chamada resumo(), que mostre na tela algumas informações geradas pelas funções que já temos no módulo criado até aqui.

Exercício Python 111: Crie um pacote chamado utilidadesCeV que tenha dois módulos internos chamados moeda e dado. Transfira todas as funções utilizadas nos desafios 107, 108 e 109 para o primeiro pacote e mantenha tudo funcionando.

Exercício Python 112: Dentro do pacote utilidadesCeV que criamos no desafio 111, temos um módulo chamado dado. Crie uma função chamada leiaDinheiro() que seja capaz de funcionar como a função imputa(), mas com uma validação de dados para aceitar apenas valores que seja monetários.

Exercício Python 113: Reescreva a função leiaInt() que fizemos no desafio 104, incluindo agora a possibilidade da digitação de um número de tipo inválido. Aproveite e crie também uma função leiaFloat() com a mesma funcionalidade.

Exercício Python 114: Crie um código em Python que teste se o site pudim está acessível pelo computador usado.

Exercício Python 115a: Vamos criar um menu em Python, usando modularização.

Exercício Python 115b: Vamos ver como fazer acesso a arquivos usando o Python.



Exercício Python 115c: Vamos finalizar o projeto de acesso a arquivos em Python.